core-backend/sandbox/src/funcs.rs

```diff
@@ -400,6 +401,25 @@ where
         })
     }
 
+    pub fn random(ctx: &mut Runtime<E>, args: &[Value]) -> SyscallOutput {
+        sys_trace!(target: "syscall::gear", "random, args = {}", args_to_str(args));
+
+        let (random_ptr, subject_ptr, subject_len, bn_ptr) = args.iter().read_4()?;
+
+        ctx.run(|ctx| {
+            let mut subject = ctx.read_memory(subject_ptr, subject_len)?;
+            subject.reserve(32);
```

Comment:
Maybe it is worth to use `LimitedVec` here and in wasmi backend?

Reply:
We can't use `LimitedVec` because subject size is unlimited.

Reply:
Yep and this is the reason `LimitedVec` was created for. We can't allocate buffers longer than some specific value.